distributed/utils.py

```diff
@@ -386,7 +386,7 @@ def key_split_group(x):
     typ = type(x)
     if typ is tuple:
         return x[0]
-    elif typ is str:
+    elif typ is str or typ is six.text_type:
```

Comment:
I'm not familiar with performance optimization in Python so I'll defer completely to you! I'll do that.